### ISSUE TITLE
Navigation does not work when mixing programmatic navigation with RouterLink navigation

### ISSUE THREAD

**Issue by igormilina** [vaadin/flow#8575]
### Description of the bug:

I have two Java view classes, for example: ViewA and ViewB. Using RouterLink component for navigation between those two views works as expected - click on (router) link on ViewA will open ViewB, and click on link on ViewB will go back to ViewA. But, if I have Button on ViewA that executes:

	UI.getCurrent().navigate(ViewB.class);

This will correctly open ViewB, but RouterLink that should lead back to ViewA will no longer work. Click on this link will set URL in browser to viewA, but view itself will not change. Also, no request to server is sent.

### Minimal reproducible example

This behaviour is reproducible in Vaadin example project (created using https://start.vaadin.com/) using any two Java views. Programmatic navigation using Button and RouterLink should be added to views as described above.

### Expected behavior

Programmatic navigation should not cause RouterLink to stop working

### Actual behavior

Programmatic navigation from one view causes RouterLink navigation (that leads back to the same view) to stop working. This is working fine in 14.1.16

### Versions:
    - Vaadin / Flow version: 16.0.0 (haven't tested with latest V14 or V15)
    - Java version: 11
    - OS version: Win10
    - Browser version (if applicable): Latest Chrome and Firefox
    - Application Server (if applicable): Tomcat9

**Comment by caalador** [comment]
Issue verified for 16, but at least doesn't replicate on 14.2.1

**Comment by sir-vince** [comment]
I also experienced the issue but RouterLink only breaks for me when I try to navigate back to the original page.
Eg.
Example 1: Frontpage -> Page1 (RouterLink) -> Page2 (Programmatic) -> Page1 (RouterLink) = No action is done but the URL is updated
Example 2: Frontpage -> Page1 (RouterLink) -> Page2 (Programmatic) -> Frontpage (RouterLink) -> Page 1 (RouterLink) = Works

Hypothesis:
I can see in "JavaScriptBootstrapUI" in the method "navigation", the js command "setTimeout(() => window.history.pushState(null, '', $0))" is issued but when checking my browser (Chrome) "window.onpopstate" is null. 

As I remember, this might break the "back" functionality but I'm not sure how to verify whether it will also break the RouterLink functionality.

**Comment by haijian-vaadin** [comment]
Hi, thanks for reporting the issue, the ticket will be fixed once [this PR] is merged(https://github.com/vaadin/platform/commit/4360f23c0b3168d484842752de008a59508f65a7)

Meanwhile, a quick workaround is to use [npm mode](https://vaadin.com/docs/v16/flow/advanced/tutorial-switch-npm-pnpm.html) instead of pnpm. 

Vaadin 17 doesn't have this issue since it's already using the latest router version.

**Comment by joheriks** [comment]
Closing as duplicate of #8071 and #8689.
